src/test/ui/nll/ty-outlives/projection-two-region-trait-bound-closure.stderr

```diff
@@ -167,17 +167,20 @@ LL | |     T: Anything<'b, 'b>,
    = note: defining type: two_regions::<'_#1r, T>
 
 error: lifetime may not live long enough
-  --> $DIR/projection-two-region-trait-bound-closure.rs:87:29
+  --> $DIR/projection-two-region-trait-bound-closure.rs:87:5
    |
 LL | fn two_regions<'a, 'b, T>(cell: Cell<&'a ()>, t: T)
    |                --  -- lifetime `'b` defined here
    |                |
    |                lifetime `'a` defined here
 ...
 LL |     with_signature(cell, t, |cell, t| require(cell, t));
-   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^ closure body requires that `'b` must outlive `'a`
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ argument requires that `'b` must outlive `'a`
```

Comment:
This is another minor diagnostic regression, but I believe it never was an intended behavior of the previous setup. Actually at the point of rendering diagnostics the intention was to not have a constraint with `ClosureBounds` category, but here we've somehow missed converting `ClosureBounds` constraints.

Anyway this regression only affect the niche case where there is double lifetime bounds in assoc type definition (`Trait<'a, 'b>::Assoc: 'a + 'b`) AND the bound `X: Trait<'a, 'a>` appears in param env.